e2e/cypress/integration/files_and_attachments/upload_files_spec.js

```diff
@@ -52,4 +54,89 @@ describe('Upload Files', () => {
+                extensions: 'JPG',
+                type: 'image',
+            },
+        ];
+
+        attachmentFilesList.forEach((file) => {
+            // # Attach the file as attachment and post a message
+            cy.get('#fileUploadInput').attachFile(file.filename);
+            cy.postMessage(MESSAGES.SMALL);
+
+            // # Get the post id of the last message
+            cy.getLastPostId().then((lastPostId) => {
+                // # Scan inside of the last post message
+                cy.get(`#${lastPostId}_message`).should('exist').and('be.visible').within(() => {
+                    // # If file type is document then file container will be rendered
+                    if (file.type === 'document') {
+                        // * Check if the download icon is exists
+                        cy.findByLabelText('download').should('exist').and((fileAttachment) => {
+                            // * Verify that download link has correct name
+                            verifyLinkHasDownloadProperties(fileAttachment, file.filename);
+                        });
+
+                        // * Check if the file name is shown in the attachment
+                        cy.findByText(file.filename).should('be.visible');
+
+                        // * Check if correct extension is shown in the attachment and click to open preview
+                        cy.findByText(file.extensions).should('be.visible').click();
+                    } else if (file.type === 'image') {
+                        // * Check that image is in the post message with valid source link
+                        cy.findByLabelText(`file thumbnail ${file.filename}`).should('be.visible').
+                            and('have.attr', 'src');
+
+                        // # Click on the image to open the preview
+                        cy.findByLabelText(`file thumbnail ${file.filename}`).click();
+                    }
+                });
+            });
+
+            // * Verify preview modal is opened
+            cy.get('.a11y__modal').should('exist').and('be.visible').
+                within(() => {
+                    // * Check for download property of the download button
+                    cy.findByText('Download').should('be.visible').parent().and((fileAttachment) => {
+                        // * Verify that download link has correct name
+                        verifyLinkHasDownloadProperties(fileAttachment, file.filename);
+                    });
+                });
+
+            // # Close the image preview modal
+            cy.get('body').type('{esc}');
+        });
+    });
 });
+
+function verifyLinkHasDownloadProperties(fileAttachment, filename) {
+    // * Verify if download attribute exists which allows to download instead of navigation
+    expect(fileAttachment.attr('download')).to.equal(filename);
+
+    // * Verify it has a not empty download link
+    expect(fileAttachment.attr('href')).is.not.null;
```

Comment:
I think we should do request to the download link and verify some of its properties, like
```suggestion
    cy.request(fileAttachment.attr('href')).then((res) => {
        expect(res.headers['content-disposition']).to.equal(`attachment;filename="${filename}"; filename*=UTF-8''${filename}`);
    });
```

Note that it will work if you do `.then((fileAttachment) => {` instead of `.and` both at [here](https://github.com/mattermost/mattermost-webapp/pull/6248/files#diff-c9a96f255e269faf4adce4e32b2b0bb7R99) and [here](https://github.com/mattermost/mattermost-webapp/pull/6248/files#diff-c9a96f255e269faf4adce4e32b2b0bb7R124).